jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -429,21 +429,35 @@ private void writeErrorPlain(HttpServletRequest request, PrintWriter writer, int
 
     private void writeErrorJson(HttpServletRequest request, PrintWriter writer, int code, String message)
     {
+        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
+        Object servlet = request.getAttribute(Dispatcher.ERROR_SERVLET_NAME);
+
         writer
             .append("{\n")
-            .append("  url: \"").append(request.getRequestURI()).append("\",\n")
-            .append("  status: \"").append(Integer.toString(code)).append("\",\n")
-            .append("  message: ").append(QuotedStringTokenizer.quote(message)).append(",\n");
-        Object servlet = request.getAttribute(Dispatcher.ERROR_SERVLET_NAME);
+            .append("  \"url\": \"").append(request.getRequestURI()).append("\",\n")
+            .append("  \"status\": \"").append(Integer.toString(code)).append("\",\n")
+            .append("  \"message\": ").append(QuotedStringTokenizer.quote(message));
+
         if (servlet != null)
-            writer.append("servlet: \"").append(servlet.toString()).append("\",\n");
-        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
+        {
+            writer.append(",\n");
+            writer.append("servlet: \"").append(servlet.toString()).append("\"");
+        }
+        if(cause!=null)
+            writer.append(',');
+
+        writer.append("\n");
+
         int c = 0;
         while (cause != null)
         {
-            writer.append("  cause").append(Integer.toString(c++)).append(": ")
-                .append(QuotedStringTokenizer.quote(cause.toString())).append(",\n");
+            writer.append("  \"cause").append(Integer.toString(c++)).append('"').append(": ")
+                .append(QuotedStringTokenizer.quote(cause.toString()));
```

Comment:
I'm not sure we want to send to the client the exception message. Isn't this a security issue, as the client could trigger an exception with a message that is then relayed back to the client? @gregw opinions?